src/cursor/cur_file.c

```diff
@@ -152,9 +159,14 @@ __curfile_equals(WT_CURSOR *a, WT_CURSOR *b, int *equalp)
     WT_CURSOR_BTREE *cbt;
     WT_DECL_RET;
     WT_SESSION_IMPL *session;
+    bool tracked;
```

Comment:
What would you think of creating three macros:
```
  #define  WT_DECL_CUR_TRACK    bool _tracked = false;
  #define WT_CUR_TRACK_BEGIN(session)  __wt_api_track_cursor_begin(session); _tracked = true
  #define WT_CUR_TRACK_END(session) if (_tracked) __wt_api_track_cursor_end(session)
```
It might appear a bit cleaner, and explanatory comments would be in one place.

Since we have a larger set of these BEGIN/END matching type macros, we could create a script in dist that checks to make sure that, when used, they match up and are ordered as expected.  (Another ticket).

Reply:
It is interesting. I'll try it out.

I moved away from the tracking macro approach since I wanted the functionality to be in a function so I could add diagnostic checks that were accessible in a debugger. Hopefully that isn't necessary after this change is merged.